tests/acceptance/test_config_refresh.py

```diff
@@ -73,3 +74,21 @@ def test_no_config_refresh_after_5_seconds(tmp_path, config):
         "Configuration version didn't change. Continue running with current version.",
         test_timeout=7,
     )
+
+
+def test_recover_after_invalid_then_valid_config(tmp_path, config):
+    config_path = tmp_path / "generated_config.yml"
+    config_path.write_text(config.as_json())
+    proc = start_logprep(config_path)
+    wait_for_output(proc, "Config refresh interval is set to: 5 seconds", test_timeout=5)
+
+    config.config_refresh_interval = None
+    config.version = "invalid"
+    config_path.write_text(config.as_json())
+    time.sleep(3)
```

Comment:
sleep actions in test leads to flaking tests because you never know if a test run on a slower system does not needs some seconds more. It is better to wait for some condition in the logs for example.